webproject/src/main/java/com/google/sps/servlets/JoinServlet.java

```diff
@@ -0,0 +1,91 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.sps.authentication.AuthenticationHandler;
+import com.google.sps.data.Message;
+import com.google.sps.data.UserRoom;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+@WebServlet("/join")
+public class JoinServlet extends HttpServlet {
+    @Override
+    public void init() {
```

Comment:
lets refactor this later, I believe we currently have 2 way of fetching/updating data to firebase